src/components/TheLoader.vue

```diff
@@ -0,0 +1,22 @@
+<script setup lang="ts">
+import { onMounted, onUnmounted, ref } from "vue";
+
+withDefaults(defineProps<{ message?: string }>(), { message: "Loading" });
```

Comment:
Just a minor formatting thing, I think it's easier to read `withDefaults` when you do separate lines like in this example from the docs
```typescript
const props = withDefaults(defineProps<Props>(), {
  msg: 'hello',
  labels: () => ['one', 'two']
})
```